pkg/certificates/ssh.go

```diff
@@ -0,0 +1,215 @@
+package certificates
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+// sshClient interface and sshDialer type remain the same.
+type sshClient interface {
+	Close() error
+	NewSession() (*ssh.Session, error)
+}
+
+// sshDialer is a function type for dialing SSH connections.
+type sshDialer func(network, addr string, config *ssh.ClientConfig) (sshClient, error)
+
+// SSHRunner provides methods for running commands over SSH.
+type SSHRunner interface {
+	// RunCommand runs a command on the remote host
+	RunCommand(ctx context.Context, node string, cmds []string) error
+	// RunCommandWithOutput runs a command on the remote host and returns the output
+	RunCommandWithOutput(ctx context.Context, node string, cmds []string) (string, error)
+}
+
+// DefaultSSHRunner is the default implementation of SSHRunner.
+type DefaultSSHRunner struct {
+	sshConfig  *ssh.ClientConfig
+	sshDialer  sshDialer
+	sshKeyPath string
+	sshPasswd  string
+}
+
+// NewSSHRunner creates a new SSH runner with the given configuration.
+func NewSSHRunner(cfg SSHConfig) (*DefaultSSHRunner, error) {
+	r := &DefaultSSHRunner{
+		sshDialer: func(network, addr string, config *ssh.ClientConfig) (sshClient, error) {
+			return ssh.Dial(network, addr, config)
+		},
+	}
+
+	r.sshKeyPath = cfg.KeyPath
+
```

Comment:
define it here so that we can use the r.sshPasswd inside parsePrivateKey instead of passing it as a param.
`r.sshPasswd = passwd`